doc/specifications/proposed/named_queries/developer_documentation.md

```diff
@@ -0,0 +1,202 @@
+# Listing Content and Locations
+
+Starting from Platform 2015.07, listing content and locations in a project is
+much easier, using a couple standardized mechanisms.
+
+Predefined, named queries can be defined in any bundle. They can then be used from twig, using a
+[custom controller](#query_controller), or from PHP code using the [QueryType Registry](#fixme). The controller supports
+[template overriding](#template_override), making it easy to use different templates for different calls.
+
+## <a name="query_controller"></a> The query controller
+A new controller is added : `ez_query`.
+
+It has two actions: `locationQuery`, and `contentQuery`.
+
+Both requires two arguments:
+- `queryName`, the name of the QueryType to run. Example: `AcmeBundle:LatestArticles'
+- `viewType`, similar to what `viewContent` and `viewLocation` expect. Examples: `list`, `tree`, `full_list`...
+
+If the QueryType supports any, they can be provided using the `parameters` argument (hash):
+
+```twig
+{{ render(controller(
+    'ez_query:locationQuery',
+    {
+        'queryName': 'AcmeBundle:LatestContent',
+        'viewType': 'list',
+        'parameters': {'ContentType': 'article'}
```

Comment:
Good idea. But in addition, I'm thinking that the query parameters could also be passed on to the template. This could be pretty useful, don't you think ?